### ISSUE TITLE
[Cms] Cannot log into different installs simultaneously 

### ISSUE THREAD

**Issue by nilshoerrmann** [getkirby/kirby#454]
It's not possible to be logged into two different installs at the same time: switching tabs always logs out the user in the previously visited site.

Maybe related: This happens with the same user and same password for the different sites.

**Comment by lukasbestle** [comment]
Are both installations on the same domain (e.g. `localhost`)?

**Comment by nilshoerrmann** [comment]
Yes, they are: `localhost`, same port.

**Comment by lukasbestle** [comment]
Ah, alright.

That‘s because the session cookies overwrite themselves on the same domain.

A workaround is to use different cookie names by configuring them in the Kirby config, see the [Session docs](https://www.notion.so/getkirby/Sessions-Concept-7d678d4d00c846ed81b3f98b71c6bc9b).

A proper fix would be if we set the cookie‘s `path` attribute to the base path of the site. The question is whether this use-case of multiple sites per domain is a common one. What do you think @bastianallgeier @distantnative?

**Comment by jenstornell** [comment]
My guess is that it's going to happen from time to time and it could really cause both frustration and additional forum support questions and Github issues.

Personally I use `localhost` because it's the default when using Xampp (like I do). It's also the default when using Wamp. These are the two biggest PHP/Apache localhost servers on Windows.

**My domain structure on localhost:**

```text
localhost/domains/some
localhost/domains/awesome
localhost/domains/sites
localhost/plugins/my
localhost/plugins/awesome
localhost/plugins/things
```

In many cases we may only work on a single domain at a time, but from time to time we need to move stuff between them. In other times we just need to compare two panels.

Because of what is expected and what is happening that causes this issue, I would flag it as a bug.

@lukasbestle I agree that using the `path` somehow is the way to go to make the cache unique. Just make sure that the path is like `localhost/domain-name` and not just `localhost`.

**In conclusion:** Having a great default setup would help so much in this case, where the user in most cases don't even need to configure the session in most cases.

**Comment by nilshoerrmann** [comment]
Thanks, @lukasbestle, for the link to the session docs and the workaround.

> The question is whether this use-case of multiple sites per domain is a common one.

Well, I guess it depends on your own habits wheather you consider this a common use-case or not.

We use MAMP for local development and adding a folder for each site we work on is the most straightforward setup for us. We use a similar setup to @jenstornell:

```
localhost/clientone.com
localhost/clienttwo.com
```

We also have a similar structure on our private project server we use to share sites with our clients during the design process:

```
ourprojectserver.com/clientone.com
ourprojectserver.com/clienttwo.com
```

> In many cases we may only work on a single domain at a time, but from time to time we need to move stuff between them. In other times we just need to compare two panels.

Exactly. This is how I noticed the issue.

I now understand why this happens and I understand the logic behind it. It's just counterintuitive from our perspective, because – for us – we separated the two Kirby instances by using different folders. Of course, both installs are still on `localhost` but Kirby binds the session to a path above our install roots.

**Comment by lukasbestle** [comment]
Your arguments make a lot of sense. I can understand that the current behavior can be quite confusing.

I will look into this. If I can find a reliable solution that works for all use-cases, I will implement that.

**Comment by distantnative** [comment]
✅
